order/views.py

```diff
@@ -1,3 +1,142 @@
-from django.shortcuts import render
+import json
+from random import randint
 
-# Create your views here.
+from django.views import View
+from django.http import JsonResponse
+
+from .models import Order, OrderList, OrderStatus
+from product.models import Product, Color, ProductColor, Size, ProductSize
+from user.models import User
+
+def delivery_fee_calculator(orderlists, current_order):
+    sum = 0
+    for orderlist in orderlists:
+        sum += orderlist.product.price*orderlist.quantity
+    if sum > 50000:
+        current_order.delivery_fee = 0
+        current_order.save()
+    else:
+        current_order.delivery_fee = 2500
+        current_order.save()
+
+class OrderListView(View):
+    def post(self, request):
+        data = json.loads(request.body)
+        try:
+            user = User.objects.get(pk=data["user_id"])
+            quantity = data["quantity"]
+            product = Product.objects.get(pk=data["product_id"])
+            color = data.get("color",0)
+            size = data.get("size",0)
+            order = Order.objects.filter(user=user, status_id=1)
+            current_order = order.last()
+            print("------order------",order)
+            if not order.exists():
+                created_order = Order.objects.create(
+                    user         = user,
+                    status_id    = 1
+            )
+                year          = str(created_order.updated_at.year)
+                month         = str(created_order.updated_at.month)
+                day           = str(created_order.updated_at.day)
+                random_number = str(randint(10000000,99999999))
+                number        = year+month+day+random_number
+                created_order.number = number
+                created_order.save()
+
+            existing_orderlist = OrderList.objects.filter(product=product, order_id=current_order.id, color=color, size=size)
+            if existing_orderlist.exists():
+                existing_orderlist.update(quantity=existing_orderlist.first().quantity + quantity)
+                existing_orderlist.first().save()
+            else:
+                OrderList.objects.create(
+                    quantity = quantity,
+                    product  = product,
+                    order    = current_order,
+                    color    = color,
+                    size     = size
+                )
+            
+            orderlists = current_order.orderlist_set.all()
+            delivery_fee_calculator(orderlists, current_order)
+            return JsonResponse({"message":"SUCCESS"}, status=200)
```

Comment:
```suggestion
            return JsonResponse({"message":"SUCCESS"}, status=201)
```
어떤 데이터가 "생성"되었을 때는 201번 상태 코드를 한 번 사용해보세요

Reply:
넵ㅎㅎ

Reply:
![image](https://user-images.githubusercontent.com/63498876/100303199-55491f00-2fdf-11eb-927a-9d97ef500308.png)